bin/resistance_marker_module.py

```diff
@@ -3,379 +3,480 @@
+            {'Codon': codon, 'FollowsIndel': follows_indel})]))
+        return marker_with_codon_info
+
+
+def pseudo_cigar_to_mutations(pseudo_cigar):
+    """Parse a pseudo CIGAR string into a list of tuples (position, operation, value)."""
+    logging.debug(f"Entering `parse_pseudo_cigar` with string={pseudo_cigar}")
+
+    if not isinstance(pseudo_cigar, str):
+        logging.error(
+            f"Expected a string for `parse_pseudo_cigar` but got {type(pseudo_cigar)}. Value: {pseudo_cigar}")
+        raise TypeError(
+            f"Expected a string for `parse_pseudo_cigar` but got {type(pseudo_cigar)}. Value: {pseudo_cigar}")
+
+    pseudo_cigar = pseudo_cigar.replace("=", "")  # Remove the "=" separator
+    logging.debug(f"Removed '=' separator. Modified string={pseudo_cigar}")
 
-    transtab = str.maketrans("TACG", "ATGC")
     tuples = []
 
     pattern = re.compile(r'(\d+)([ID\+]?)\+?(\d+|[ACGTN]+)?')
-    matches = pattern.findall(string)
+    matches = pattern.findall(pseudo_cigar)
     logging.debug(f"Pattern found {len(matches)} matches in the string")
 
     for match in matches:
-        position, operation, value = match 
-        position = refseq_len - int(position) if orientation == "-" else int(position) - 1 # use base 0 indexing
-        logging.debug(f"Processing match: original position={match[0]}, adjusted position={position}, operation={operation}, value={value}")
+        position, operation, value = match
+        logging.debug(
+            f"Processing match: original position={match[0]}, adjusted position={position}, operation={operation}, value={value}")
 
         # Handle mask
-        if operation == "+":  
+        if operation == "+":
             logging.debug("Operation is a mask. No changes applied.")
-            pass # do nothing for now
-        else:     
-            value = value.translate(transtab) if orientation == "-" else value
-            tuples.append((position, None if operation == '' else operation, value))
-            logging.debug(f"Added tuple to result: {(position, None if operation == '' else operation, value)}")
+            pass  # do nothing for now
+        else:
+            tuples.append((position, None if operation ==
+                          '' else operation, value))
+            logging.debug(
+                f"Added tuple to result: {(position, None if operation == '' else operation, value)}")
 
-    logging.debug(f"Exiting `parse_pseudo_cigar` with {len(tuples)} tuples generated")
+    logging.debug(
+        f"Exiting `parse_pseudo_cigar` with {len(tuples)} tuples generated")
     return tuples
 
 
+def extract_mutations_from_unique_pseudo_cigar(df, ref_sequences):
+    """Extracts mutations from pseudo CIGAR strings in dataframe."""
+    logging.info("Extracting mutations from unique pseudo CIGAR strings.")
+    results = []
+    transtab = str.maketrans("TACG", "ATGC")
+    for index, row in df.iterrows():
```

Comment:
I didn't, but it's already so fast I don't think it's really worth it

Reply:
how large of a dataset did you use? I agree this is probably not going to be impacted greatly as sample sizes grow since I believe it's taking unique asv and reference combinations and will therefore be more dependent on dataset noise / panel size

Reply:
I ran it on 9 runs all together and it took 3 seconds